public/app/plugins/datasource/influxdb/datasource.ts

```diff
@@ -157,7 +157,7 @@ export default class InfluxDatasource {
     return false;
   };
 
-  metricFindQuery(query) {
+  metricFindQuery(query: string, db?: string) {
```

Comment:
the second param is already taken, not sure it's used yet but we plan to. Needs to be an object (for query options). For example, it's where Grafana can add time range limits to this function and make other specific data source option (if we have specific UI for template variables for example). 

So please change this to an optional object param named options. And on that have a property named database.